### PR TITLE
FR: Lower the ISO 12646 border size

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
There have been discussions and complains of several users about the ISO 12646 white and grey border sizes being a) too large and b) too small for the grey outer part.

As there is no "you have to do so" reference and workflows / requiremenmts seem to differ this pr implements via 2 hidden conf keys (so can be user changed)

1. A minimal width of border (grey plus white part) including some safety checks for very small windows / screens
2. A ratio of the white part

As
a) the 4.2.0 border seemed to be a bit large
b) the 4.0 border was a bit too small
c) the 4.2.0 grey part was relatively small

i propose as defaults
1. a total border width of 4cm
2. with 40% of white

### CODE REVIEW

**Comment by jenshannoschwalm** [comment]
@Nilvus and @gi-man any other suggestions as defaults?

@TurboGit I guess this could be also for 4.2.1 as almost no code is touched.

File: data/darktableconfig.xml.in
Comment (TurboGit) [inline data/darktableconfig.xml.in:2129]: can't we define some min / max here to avoid crazy values possible breaking the display?

**Review by TurboGit — CHANGES_REQUESTED** [review]
Also when zooming while the iso-12646 borders are displayed there is a strange effect not visible on master. During the zoom the borders are somehow larger/expanded and when zooming stop the border is adjusted again at the right size.

**Comment by Nilvus** [comment]
> @Nilvus and @gi-man any other suggestions as defaults?

I think we should do the opposite: set minimal grey border (ie: first respect border size for grey around image, set on darkroom tab in main preferences), then set white border, with at least 1cm size and max to 1inch (2,54cm). Of course, this will remains to be tested. That what I suggested on my comment on discussion we had. See: https://github.com/darktable-org/darktable/issues/13386#issuecomment-1398327627. On that comment, I suggest to set first grey around because on color specifications shown, grey around is always higher than white size (I talk about twice but when I see that on color website, it's about grey size between 2 images with white border, so if one, we could have same grey size as white size but no less of course).

**Comment by Nilvus** [comment]
And I also agree here with @TurboGit

**Comment by gi-man** [comment]
I found using 2.5 for the border and 0.5 for the ratio looked the best (with portrait and landscape images). I'm using a 27in monitor with a 2560 x 1440 resolution at 125% scaling. I measured using a ruler at full screen settings and it was 14mm from the top bar to the edge of the white border and 14mm from the edge of the white to the edge of the image. 

Like Pascal, I noticed that using a smaller darktable screen size makes the border go larger. But this effect is only if I reduce the height of the window. If I reduce the width to the minimum darktable allows, then the border size remains consistent with full screen.

File: src/develop/develop.c
Comment (TurboGit) [inline src/develop/develop.c:774]: To avoid mixing double/float : 2.54f

File: src/views/darkroom.c
Comment (TurboGit) [inline src/views/darkroom.c:1505]: Likewise 2.54f

File: data/darktableconfig.xml.in
Comment (TurboGit) [inline data/darktableconfig.xml.in:2131]: so default is 4.0 cm, but that's not what I'm seeing.

I think this is due to:

`tb = MIN(MAX(2, bsize), 0.3f * MIN(wd, ht));`

**Review by TurboGit — CHANGES_REQUESTED** [review]
Some comments and changes.

File: data/darktableconfig.xml.in
Comment (jenshannoschwalm) [reply data/darktableconfig.xml.in:2131]: That just ensures we have at least a total border size of 2 but will never be greater than 30% of width or height.

File: src/develop/develop.c
Comment (jenshannoschwalm) [reply src/develop/develop.c:774]: Was not sure as the dpi is a double ??

**Comment by jenshannoschwalm** [comment]
Sorry i didn't explain the latest commits and my approach, had to rush to work.

Latest commits fixed the flickering bug, added two symbolic colors and takes care for hires monitors (ppd != 1)

How are conf parameters handled.
1. The "iso_border" size (default = 4cm) is the requsted border **total** width, including the white and grey parts.
2. The ratio defines the white part's size, so for a total border of 4cm the default for white woule be 40% (or whatever we agree on) of 4cm (1.6cm) Don't you see that?

I will go for whatever we agree on:
1. minimal total border size would be 2.5 cm?
2. max could be 10cm?
3. for the ration we should have a default of less than 0.5?

File: src/develop/develop.c
Comment (TurboGit) [reply src/develop/develop.c:774]: Ah its a double... so indeed that's not an issue.

**Comment by TurboGit** [comment]
iso12464_border

> 2\. The ratio defines the white part's size, so for a total border of 4cm the default for white woule be 40% (or whatever we agree on) of 4cm (1.6cm) Don't you see that?

I see exactly that, that's just that I was confused by the fact that this is total border size.

So to me this is good to go, just a pending comment about possible min/max for `iso12464_border`. Do you think we can put in place some bounds there?

**Comment by jenshannoschwalm** [comment]
I can do some senseful min and max.

Just would be interested in @Nilvus opinion about the defaults.

File: data/darktableconfig.xml.in
Comment (jenshannoschwalm) [reply data/darktableconfig.xml.in]: Did that

**Comment by jenshannoschwalm** [comment]
Combined commits for blaming and did a min/max for border width.

Fine for me ...

**Comment by TurboGit** [comment]
> Fine for me ...

I'll let @Nilvus review, but at this stage we can easily change the sizes so the important part is working in this PR. Thanks!

**Comment by Nilvus** [comment]
New defaults are great to me and much much better than what we had. Thanks. As @gi-man made a previous proposal, I think it would be good if we good have his feedback on that PR.

Just a thing remains not great: transition when zooming. Could this be smoother? Transition are a little hard, with some little flickering remaining. I tested that with your last commits [15343ef](https://github.com/darktable-org/darktable/pull/13441/commits/15343efbd35fac7affe55136da49d63056484b5a)

**Comment by TurboGit** [comment]
> Just a thing remains not great: transition when zooming. Could this be smoother? Transition are a little hard, with some little flickering remaining. I tested that with your last commits [15343ef](https://github.com/darktable-org/darktable/pull/13441/commits/15343efbd35fac7affe55136da49d63056484b5a)

We have the same on master I think. So not a regression.

**Comment by gi-man** [comment]
I tested on my system. Using 4.0 and 0.4, I get 26mm from edge to white and 18mm from white to image, therefore a total of 44mm. I think merging with these defaults is good. Since it is adjustable, users can tweak to their displays. I might keep mine at 3.0, 0.4.

**Comment by jenshannoschwalm** [comment]
Will try too get fixed let me think...

**Comment by jenshannoschwalm** [comment]
I tried to fix the remaining flickering (the reason for this is we don't know in the preview pipe if the aspects for darkroom & preview size do fit and how to setup the white border) issue. If i find time and have a good idea i will come back but for now i would prefer to get this merged as "good enough"

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**
